util/benchmark/ec/p256_awslc_ossl/src/benchmark.c

```diff
@@ -0,0 +1,98 @@
+/*
+------------------------------------------------------------------------------------
+ Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ SPDX-License-Identifier: Apache-2.0
+------------------------------------------------------------------------------------
+*/
+
+#include <stdio.h>
+#include <time.h>
+#include <assert.h>
+#include <unistd.h>
+#include <sys/times.h>
+#include <string.h>
+
+#include "benchmark.h"
+
+BIO *bio_out = NULL;
+BIO *bio_err = NULL;
+
+void open_test_streams(void)
+{
+    bio_out = BIO_new_fp(stdout, BIO_NOCLOSE);
+    bio_err = BIO_new_fp(stderr, BIO_NOCLOSE);
+
+    assert(bio_out != NULL);
+    assert(bio_err != NULL);
+}
+
+void close_test_streams(void)
+{
+    (void)BIO_flush(bio_out);
+    (void)BIO_flush(bio_err);
+
+    BIO_free_all(bio_out);
+    BIO_free_all(bio_err);
+}
+
+uint64_t time_now(void)
+{
+    struct timespec ts;
+    uint64_t ret = 0;
+
+#if defined(AARCH64_COUNTER_TIMER)
```

Comment:
I'm not sure why we need the separate branch for ARM, is clock_gettime() not working on ARM?